src/systemd/sd-id128.h

```diff
@@ -126,11 +126,16 @@ _sd_pure_ static __inline__ int sd_id128_is_null(sd_id128_t a) {
         return a.qwords[0] == 0 && a.qwords[1] == 0;
 }
 
+_sd_pure_ static __inline__ int sd_id128_is_firmware(sd_id128_t a) {
+        return a.qwords[0] == 0 && a.qwords[1] == 1;
+}
+
 _sd_pure_ static __inline__ int sd_id128_is_allf(sd_id128_t a) {
         return a.qwords[0] == UINT64_C(0xFFFFFFFFFFFFFFFF) && a.qwords[1] == UINT64_C(0xFFFFFFFFFFFFFFFF);
 }
 
 #define SD_ID128_NULL ((const sd_id128_t) { .qwords = { 0, 0 }})
+#define SD_ID128_FIRMWARE ((const sd_id128_t) { .qwords = { 0, 1 }})
```

Comment:
no, please don't. 

sd-id128 is a public C API providing a super generic id128/uuid interface, it should not have a global identifier for a special case in the machine ID initialization code.